server/game/player.js

```diff
@@ -200,11 +203,26 @@ class Player extends Spectator {
     }
 
     modifyGold(amount) {
-        this.gold += amount;
+        var goldGain = amount;
 
-        if(this.gold < 0) {
-            this.gold = 0;
+        if(amount > 0) {
+            if(!this.canGainGold()) {
+                this.addMessage('{0} cannot gain gold', this);
```

Comment:
After the gold encapsulation PR (sorry), this is no longer an appropriate place to check for `canGainGold`. 'Gaining' gold (collecting income during marshalling, Tyrion's ability, etc) is different than 'moving' gold (taking gold off of Lannisport Treasury, stealing gold with Meager Contribution, etc). Both `addGold` and `transferGold` use `modifyGold` in their underlying implementations, so this change will also inappropriately prevent moving gold.

Thus, most of this code should be moved to `Game.addGold` instead.

Reply:
Ah, I was out of luck then, I moved it to the wrong place when rebasing on top of your gold encapsulation, rather than leaving where it already was :-) Moved back now.